### PR TITLE
rec: Introduce a way to completely disable root-refresh

### LINKED ISSUE #12848: rec: Completely disable root-refresh
rec tries to resolve the `. NS` records periodically, even if `root-hints=no` (which only disables to the first step of the bootstrap to resolve `. NS` records). For forward everything scenarios. it would be nice to disable the periodic root-refresh. Current workaround is to provide a hint file referring to the forward target (which should answer `. NS` queries).

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Fixes #12848

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

@SrX please take a look, thanks!

### CODE REVIEW

File: pdns/recursordist/docs/appendices/FAQ.rst [hunk 1]
```diff
@@ -66,19 +66,23 @@ Handling of root hints
...
 
-There is another detail: after refreshing the root records, the :program:`Recursor` will resolve the ``NS`` records for the top level domain of the root servers.
+The root hints and resolved root data can differ if the root hints are outdated.
+As long as at least one root server mentioned in the root hints can be contacted, the periodic refresh will produce the desired record sets corresponding to the current up-to-date root server data.
+
+Starting with version 4.6.2, if :ref:`setting-hint-file` is set to ``no``, the :program:`Recursor` will not prime the cache with root data obtained from hints, but will still do the periodic refresh.
+A (recursive) forward configuration is be needed to make the periodic refresh work.
+
+Starting with version 4.9, setting :ref:`setting-hint-file` to ``no-refresh`` disables both the initial reading of the hints and the periodic refresh of cached root data.
+This prevents :program:`Recursor` from resolving names by itself, so it is only useful in cases where all queries are forwarded.
```
Comment (fredmorcos) [inline pdns/recursordist/docs/appendices/FAQ.rst:83]: This prevents *the* Recursor

**Review by fredmorcos — APPROVED** [review]

File: pdns/recursordist/docs/appendices/FAQ.rst (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/appendices/FAQ.rst:83]: I think the usage without article is right for a product name. We're (slowly) moving to that style form in other parts of the docs as well.

File: pdns/recursordist/docs/appendices/FAQ.rst [hunk 2]
```diff
@@ -66,19 +66,23 @@ Handling of root hints
 On startup, the :program:`Recursor` uses root hints to resolve the names and addresses of the root name servers and puts the record sets found into the record cache.
 This is needed to be able to resolve names, as the recursive algorithm starts at the root (using cached data) and then tries to resolve delegations until it finds the name servers that are authoritative for the domain in question.
 
-If the :ref:`setting-hint-file` is not set, it wil use a compiled-in table as root hints.
-Starting with version 4.6.2, if :ref:`setting-hint-file` is set to ``no``, the :program:`Recursor` will not fill the cache with root data.
-This can be used in special cases, e.g. when all queries are forwarded.
+If the :ref:`setting-hint-file` is not set, :program:`Recursor` wil use a compiled-in table as root hints.
```
Comment (rgacogne) [inline pdns/recursordist/docs/appendices/FAQ.rst:69]: s/wil/will/

File: pdns/recursordist/docs/appendices/FAQ.rst [hunk 3]
```diff
@@ -66,19 +66,23 @@ Handling of root hints
...
 If that does not succeed, it wil fall back to using the root hints to fill the cache with root data.
 Prior to version 4.7.0, the period for re-fetching root data was :ref:`setting-max-cache-ttl` divided by 12, with a minimum of 10 seconds.
 Starting with version 4.7.0, the period is adaptive, starting at 80% of :ref:`setting-max-cache-ttl`, reducing the interval on failure.
 
-There is another detail: after refreshing the root records, the :program:`Recursor` will resolve the ``NS`` records for the top level domain of the root servers.
+The root hints and resolved root data can differ if the root hints are outdated.
+As long as at least one root server mentioned in the root hints can be contacted, the periodic refresh will produce the desired record sets corresponding to the current up-to-date root server data.
+
+Starting with version 4.6.2, if :ref:`setting-hint-file` is set to ``no``, the :program:`Recursor` will not prime the cache with root data obtained from hints, but will still do the periodic refresh.
+A (recursive) forward configuration is be needed to make the periodic refresh work.
```
Comment (rgacogne) [inline pdns/recursordist/docs/appendices/FAQ.rst:80]: "is be needed" -> "is needed"?

**Comment by omoerbeek** [comment]
Thanks, I'll fix these docs issue together with the upcoming beta1 changelog commit.

**Pull request merged**